aws/resource_aws_api_gateway_authorizer.go

```diff
@@ -181,6 +199,12 @@ func resourceAwsApiGatewayAuthorizerUpdate(d *schema.ResourceData, meta interfac
 			Value: aws.String(d.Get("identity_validation_expression").(string)),
 		})
 	}
+	if d.HasChange("provider_arns") {
```

Comment:
This can be simplified using `schema.Set` `Difference()`, here's an example in action:

https://github.com/terraform-providers/terraform-provider-aws/blob/90e2e8df03e4ded02c825169bc64b3dae747b9a0/aws/resource_aws_api_gateway_integration.go#L338-L361

Reply:
Nice:+1: